Sources/NIOSSH/SSHPacketParser.swift

```diff
@@ -121,18 +123,31 @@ struct SSHPacketParser {
         let carriageReturn = UInt8(ascii: "\r")
         let lineFeed = UInt8(ascii: "\n")
 
-        // Search for version line, which starts with "SSH-". Lines without this prefix may come before the version line.
-        var slice = self.buffer.readableBytesView
-        while let lfIndex = slice.firstIndex(of: lineFeed), lfIndex < slice.endIndex {
-            if slice.starts(with: "SSH-".utf8) {
-                // Return all data upto the last LF we found, excluding the last [CR]LF.
-                slice = self.buffer.readableBytesView
+        // Per RFC 4253 §4.2:
+        //  The server MAY send other lines of data before sending the version string.
+        // This means that server does not expect any lines before version so we will return all data before first line feed
+        if self.isServer {
+            // Looking for a string ending with \r\n
+            let slice = self.buffer.readableBytesView
```

Comment:
Why are we dropping the requirement that the version line starts with `SSH-` on the server?

Reply:
in server mode we require that client send no data before `SSH`, no need to look for it, we have a check later (and a test) that checks that version starts with `SSH-`